src/backcompat.ts

```diff
@@ -0,0 +1,73 @@
+import { Hover, MarkupContent, MarkupKind, Range } from 'sourcegraph'
+import { Hover as VSCodeHover } from 'vscode-languageserver-types'
+
+// Codeintellify doesn't yet support the new, non-deprecated MarkupContent for Hover contents. The problem is that
+// when it syntax-highlights the Markdown, it sanitizes the syntax-highlighting colors away, so the code is
+// monochromatic.
```

Comment:
I am pretty sure it does support it, I took care to support the latest spec when I wrote it. It actually even converts all plain `string` `MarkedString`s (which is a very common response from language servers) to `MarkupContent`.

https://sourcegraph.sgdev.org/github.com/sourcegraph/codeintellify@5eef578cfc508dbcc6a97f5ba08675768b0a5187/-/blob/src/HoverOverlay.tsx#L137-162

If the syntax highlighting gets sanitized, all that's needed to fix is to whitelist the syntax highlighting when passing to `sanitize-html`: https://sourcegraph.sgdev.org/github.com/sourcegraph/codeintellify@5eef578cfc508dbcc6a97f5ba08675768b0a5187/-/blob/src/helpers.ts#L95-102

This should be a quick fix so I would recommend doing that directly instead of hacks to circumvent it :)

A test for `renderMarkdown` would also be helpful to prevent regressions and be quick to write.

Reply:
Can you help here and figure out what options can be passed to sanitize to make markdown code blocks work? That would be very helpful (I want to get this out soon and could use the help).